hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/JsonMetadataMutationObserver.java

```diff
@@ -46,23 +47,26 @@ public JsonMetadataMutationObserver(SerializationService serializationService,
     }
 
     @Override
-    public void onClear() {
-        // no-op
+    public void onPutRecord(@Nonnull Data key, Record record, Object oldValue, boolean backup) {
+        onPutInternal(record);
     }
 
     @Override
-    public void onPutRecord(Data key, Record record) {
+    public void onReplicationPutRecord(@Nonnull Data key, @Nonnull Record record, boolean backup) {
         onPutInternal(record);
+
     }
 
     @Override
-    public void onReplicationPutRecord(Data key, Record record) {
-        onPutInternal(record);
+    public void onUpdateRecord(@Nonnull Data key, @Nonnull Record record,
```

Comment:
Minor: maybe `oldValue`, `newValue` is a more common order. Here and are other places `oldValue` too.